From 984c0f316e6d5505d0519838246bb2463108cf1c Mon Sep 17 00:00:00 2001
From: Amir Ayub <dev.amirayub@gmail.com>
Date: Wed, 23 Sep 2026 23:40:23 +0300
Subject: [PATCH] Identify Apple silicon GPUs

Add VENDOR_APPLE, 0x106b, Apple's PCI vendor ID, and map the OpenCL vendor
ID 0x1027f00 that Apple silicon GPUs report to it.
---
 src/cbang/hw/GPUVendor.h       | 1 +
 src/cbang/hw/OpenCLLibrary.cpp | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/cbang/hw/GPUVendor.h b/src/cbang/hw/GPUVendor.h
index ee8e8498..d9011dde 100644
--- a/src/cbang/hw/GPUVendor.h
+++ b/src/cbang/hw/GPUVendor.h
@@ -47,6 +47,7 @@
 CBANG_ENUM_VALUE(VENDOR_UNKNOWN, 0)
 CBANG_ENUM_VALUE(VENDOR_AMD,     0x1002)
 CBANG_ENUM_ALIAS(VENDOR_ATI,     VENDOR_AMD)
+CBANG_ENUM_VALUE(VENDOR_APPLE,   0x106b)
 CBANG_ENUM_VALUE(VENDOR_NVIDIA,  0x10de)
 CBANG_ENUM_VALUE(VENDOR_INTEL,   0x8086)
 
diff --git a/src/cbang/hw/OpenCLLibrary.cpp b/src/cbang/hw/OpenCLLibrary.cpp
index 12dfaf30..533387a6 100644
--- a/src/cbang/hw/OpenCLLibrary.cpp
+++ b/src/cbang/hw/OpenCLLibrary.cpp
@@ -335,6 +335,7 @@ int32_t OpenCLLibrary::getVendorID(void *device) {
 
   // Integrated AMD cards on Apple return wrong vendor ID
   if (vendorID == 0x1021d00) vendorID = 0x1002;
+  if (vendorID == 0x1027f00) vendorID = GPUVendor::VENDOR_APPLE;
 
   return vendorID;
 }
-- 
2.55.0

